src/tests/Common/helixpublishwitharcade.proj

```diff
@@ -429,61 +503,86 @@
       <TestGroup Condition=" $([MSBuild]::ValueOrDefault(%(XUnitWrapperDlls), '').IndexOf('.XUnitWrapper.dll')) != $([MSBuild]::ValueOrDefault(%(XUnitWrapperDlls), '').LastIndexOf('.XUnitWrapper.dll')) "></TestGroup>
       <PayloadZipFile>$(LegacyPayloadsRootDirectory)\%(PayloadGroup).zip</PayloadZipFile>
     </LegacyPayloads>
+  </ItemGroup>
 
+  <ItemGroup>
     <MergedPayloads Include="$([System.IO.Directory]::GetDirectories($(MergedPayloadsRootDirectory)))" Condition="Exists('$(MergedPayloadsRootDirectory)')" />
 
     <MergedPayloads Update="@(MergedPayloads)">
       <PayloadDirectory>%(FullPath)</PayloadDirectory>
-      <PayloadHelixCommandFile>$(MergedPayloadsRootDirectory)%(FileName)\HelixCommand.txt</PayloadHelixCommandFile>
     </MergedPayloads>
+  </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetsMobile)' != 'true'">
     <MergedPayloads Update="@(MergedPayloads)">
+      <PayloadHelixCommandFile>$(MergedPayloadsRootDirectory)%(FileName)%(Extension)\HelixCommand.txt</PayloadHelixCommandFile>
       <MergedTestHelixCommand>$([System.IO.File]::ReadAllText('%(PayloadHelixCommandFile)'))</MergedTestHelixCommand>
```

Comment:
Does this actually work for you? When I was working on the initial implementation of the merged test wrappers, it seemed to me that within the ItemGroup Update loop you only see the pre-existing metadata values i.o.w. to consume a previously calculated metadata I had to use two subsequent Update loops. It can easily be the case I was doing something wrong, I don't have any exact documentation of this behavior at my disposal, just please double-check this truly behaves the way you expect it to.

Reply:
I'll check if this actually works and revert it if it doesn't. I was seeing success, but I'll keep an eye on the WASM leg to make sure.

Reply:
I've validated that this works correctly.